test/opsem2/trackingmem/freed_unsat.c

```diff
@@ -0,0 +1,21 @@
+//; RUN: %sea "%s" --horn-bv2-extra-widemem --horn-bv2-tracking-mem 2>&1 | OutputCheck %s
+// CHECK: ^unsat$
+// This test mainly checks that we can write multiple metadata about tracking
+// memory independently of each other.
+#include "seahorn/seahorn.h"
+#include <stdlib.h>
+
+extern void sea_reset_modified(char *);
+extern bool sea_is_alloc(char *);
+extern void sea_tracking_on();
+extern void sea_tracking_off();
+
+extern int nd_int();
+int main() {
+  sea_tracking_on();
+  int *a = malloc(1024);
+  *a = nd_int();
+  free(a);
```

Comment:
I missed where `free` is mapped to `sea.free`...

Reply:
we should not map `free` to `sea.free`. They have different semantics... This also interferes with other LLVM passes that are aware of semantics of `free`. Note that there is also `delete` in C++ that should be treated similarly.

Reply:
The semantics of `sea.free()` is to mark a memory region as freed. The memory region itself, is not freed until the call to `free()`. The semantics of `free()` is to release the address space to be allocated again.

We can, for example, provide an implementation of `free()` that first calls `sea.free()` to mark memory as freed, and then calls the builtin `free()`.  We do similar things for functions like `memcmp` that have builtin versions.

The reason why we cannot change `free()` to any other name is that both `malloc()` and `free()` are understood by LLVM. That is, they are builtin functions with well defined semantics.

To me, `sea.free` is more like `llvm.lifetime.end`, if that analogy helps.

Reply:
If changing `free` to `sea.free` is blocking this change then we need to discuss this face to face. Otherwise, I can submit it. The cause of build failure is well understood and will be fixed with next commit. I want to send a final cleanup commit once the above discussion is concluded.

Reply:
Added a `sea_free` user facing function which maps to `sea.free` builtin.